roseau/load_flow/models/transformers/transformers.py

```diff
@@ -131,39 +174,70 @@ def __init__(
-        self._tap = value
-        self._invalidate_network_results()
+        if phases2 is None:
+            phases2 = "".join(p for p in bus1.phases if p in bus2.phases)  # can't use set because order is important
+            phases2 = phases2.replace("ac", "ca")
+            if len(phases2) != 2:
+                msg = f"Phases (2) of transformer {id!r} can not be deduced from the buses, they need to be specified."
+                logger.error(msg)
+                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_PHASE)
+        else:
+            self._check_phases(id, allowed_phases=self._allowed_phases_single, phases2=phases2)
+            self._check_bus_phases(id, bus2, phases2=phases2)
 
-    @property
-    def parameters(self) -> TransformerParameters:
-        """The parameters of the transformer."""
-        return self._parameters
+        return phases1, phases2
 
-    @parameters.setter
-    def parameters(self, value: TransformerParameters) -> None:
-        windings1 = self._parameters.windings
-        windings2 = value.windings
-        if windings1 != windings2:
-            msg = f"The updated windings changed for transformer {self.id!r}: {windings1} to {windings2}."
-            logger.error(msg)
-            raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_TRANSFORMER_WINDINGS)
-        self._parameters = value
-        self._invalidate_network_results()
+    def _compute_phases_split(
+        self, id: Id, bus1: Bus, bus2: Bus, phases1: Optional[str], phases2: Optional[str]
+    ) -> tuple[str, str]:
+        if phases1 is None:
+            phases1 = "".join(p for p in bus2.phases if p in bus1.phases and p != "n")
+            phases1 = phases1.replace("ac", "ca")
+            if len(phases1) != 2:
+                msg = f"Phases (1) of transformer {id!r} can not be deduced from the buses, they need to be specified."
+                logger.error(msg)
+                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_PHASE)
+        else:
+            self._check_phases(id, allowed_phases=self._allowed_phases_single, phases1=phases1)
+            self._check_bus_phases(id, bus1, phases1=phases1)
 
-    def to_dict(self) -> JsonDict:
-        return {**super().to_dict(), "params_id": self.parameters.id, "tap": self.tap}
+        if phases2 is None:
+            phases2 = "".join(p for p in bus2.phases if p in bus1.phases or p == "n")
+            if len(phases2) != 3:
+                msg = f"Phases (2) of transformer {id!r} can not be deduced from the buses, they need to be specified."
+                logger.error(msg)
+                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_PHASE)
+        else:
+            self._check_phases(id, allowed_phases=self._allowed_phases_split_secondary, phases2=phases2)
+            self._check_bus_phases(id, bus2, phases2=phases2)
```

Comment:
In the second option `phases2 is not None`, you ensure that the neutral is present in the secondary phases using the `self._allowed_phases_split_secondary` variable (all the options have a neutral). Not in the first option `phases2 is None`. If I provide `bus1.phases="abcn"` and `bus2.phases="abc"`, I get `phases2="abc"` and `len(phases2)==3`. As far as I remember, the neutral is mandatory for split phase transformer (secondary).

Reply:
Indeed, good catch. The check with `len` was ok for single-phase, but not here.